lapis/storage.py

```diff
@@ -0,0 +1,81 @@
+from usim import time
+
+from typing import Optional
+
+from lapis.utilities.cache_algorithm_implementations import cache_algorithm
+from lapis.utilities.cache_cleanup_implementations import cache_cleanup
+
+
+class Storage(object):
+
+    __slots__ = ("name", "sitename", "storagesize", "usedstorage", "content")
+
+    def __init__(
+        self, name: str, sitename: str, storagesize: int, content: Optional[dict] = None
+    ):
+        self.name = name
+        self.sitename = sitename
+        self.storagesize = storagesize
+        self.content = content
+        self.usedstorage = self.get_used_storage()
+        self.describe_state()
+
+    def get_used_storage(self):
+        return sum(subdict["usedsize"] for subdict in self.content.values())
+
+    def free_space(self):
+        return self.storagesize - self.usedstorage
+
+    def place_new_file(self, filerequest: tuple):
+        filename, filespecs = filerequest
+        if self.free_space() - filespecs["usedsize"] < 0:
+            self.make_room(self.free_space() - filespecs["usedsize"])
+        self.content.update({filename: filespecs})
+        self.content[filename].update(
+            cachedsince=time.now, lastaccessed=time.now, numberofaccesses=0
+        )
+        self.usedstorage = self.get_used_storage()
```

Comment:
Some thoughts coming to my mind:

* The method is called `place_new_file` so I think you assume the file is not existing in the current storage. Please make an assertion about this. 
* Further it seems counterintuitive, that you do an update of the `content` dict although you consider the file being new. What is the reasoning behind?
* Why doesn't it take time to actually place the file into the storage?

Reply:
In this case updating the content dict adds the file to storage.content dict. It does the right thing but sounds counter intuitive because I use the dict.update method. I'll change it. 
How would you determine/estimate how long it takes to place a file into the storage?

Reply:
> How would you determine/estimate how long it takes to place a file into the storage?

In the first implementation I would just assume a fixed time interval. Later we can make it dependent on the available bandwidth and size to be transferred. Make it as simple as possible in the beginning.